src/core/thread/mle.cpp

```diff
@@ -4015,7 +4031,14 @@ void Mle::HandleChildUpdateResponse(RxInfo &aRxInfo)
         switch (Tlv::Find<TimeoutTlv>(aRxInfo.mMessage, timeout))
         {
         case kErrorNone:
-            mTimeout = timeout;
+            if (timeout == 0 && mDetachGracefullyTimer.IsRunning())
```

Comment:
Can timeout be 0 in any case other than detaching gracefully?

Reply:
When the build config minimal timeout is set to 0 and the user calls API to set timeout to 0.

Reply:
Should we call `Stop()` if only "build config minimal timeout is set to 0" ?

Reply:
We need to call Stop() no matter if the config minimum timeout is 0 because it’s in the process of detaching. For the config, consider 2 cases:

1. The minimal timeout is not 0: if timeout is 0 then the request must have been sent by "detach gracefully".
2. The minimal timeout is 0: if the timer isn't running, then the request must have been sent by user, we don't call Stop(); if the timer is running, the request could be either from user or "detach gracefully", but we don't care as we have done the "goodbye" message and it's ready to call Stop().